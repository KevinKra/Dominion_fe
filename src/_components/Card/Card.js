import React from "react";
import "./Card.scss";

export default function Card({ name, desc, category, image, cost, tags, id }) {
  const handleRender = () => {
    if (category !== undefined) {
      if (category[0] === "Action") {
        return (
          <article className='card' id={id} data-name={name}>
            <header style={{ backgroundColor: "silver" }}>{name}</header>
            {image ? (
              <img src={image} alt='' />
            ) : null}
            <section className='card-details-section'>
              <div className='card-details'>
                {tags.map((tag, i) => (
                  <p className='tag' key={i}>
                    {tag}
                  </p>
                ))}
                <p className='description'>{desc}</p>
              </div>
            </section>
            <footer>
              <p>{cost}</p>
              <p>{category}</p>
            </footer>
          </article>
        );
      } else if (category[0] === "Treasure") {
        return (
          <article className='card' id={id} data-name={name}>
            <header style={{ backgroundColor: "gold" }}>{name}</header>
            {image ? (
              <img
                src={image}
                alt={name}
                style={{ flex: 1 }}
              />
            ) : null}
            <footer>
              <p>{cost}</p>
              <p>{category}</p>
            </footer>
          </article>
        );
      } else {
        return (
          <article className='card' id={id} data-name={name}>
            <header style={{ backgroundColor: "lightGreen" }}>{name}</header>
            {image ? (
              <img
                src={image}
                alt={name}
                style={{ flex: 1 }}
              />
            ) : null}
            <footer>
              <p>{cost}</p>
              <p>{category}</p>
            </footer>
          </article>
        );
      }
    } else {
      return <p>Loading</p>;
    }
  };

  const cardType = handleRender();
  return cardType;
}
