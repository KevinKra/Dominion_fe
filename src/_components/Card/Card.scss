@import "../../styles/_variables.scss";
@import "../../styles/_animations.scss";

.card {
  position: relative;
  z-index: 2;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid $color1;
  border-radius: 4px;
  box-shadow: 1px 1px 3px $color2;
  height: 190px;
  width: 130px;
  min-width: 130px;
  margin: 0.3rem;
  background-color: #eee;
  transition: 0.5s all cubic-bezier(0.71, 2, 0.68, 0.79);
  animation: fadeIn 1.1s;
  .card-details-section {
    border-top: 1px solid $color2;
    height: 100%;
    width: 100%;
  }
  .card-details {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
    height: 100%;
    width: 92%;
    margin: 0 auto;
  }
  .tag {
    font-family: "Germania One", cursive;
    font-weight: 300;
  }
  .card-X {
    margin: auto;
  }
  .card-cost {
    color: gold;
  }
  img {
    height: 45%;
    width: 100%;
  }
  p {
    font-size: 9px;
    margin: 0;
    padding: 0;
  }
  header {
    font-family: "Germania One", cursive;
    font-size: 16px;
    width: 100%;
    text-align: center;
    border-top-left-radius: 2px;
    border-top-right-radius: 2px;
    border-bottom: 1px solid $color2;
  }
  footer {
    display: flex;
    justify-content: space-around;
    width: 100%;
    height: 12px;
    color: $color1;
    background-color: $color2;
    border-top: 1px solid $color2;
    border-bottom-left-radius: 4px;
    border-bottom-right-radius: 4px;
  }
  footer > p {
    font-family: "Germania One", cursive;
  }
  &:hover {
    z-index: 3;
    cursor: pointer;
    -webkit-box-shadow: 0px -1px 12px 2px $color2;
    -moz-box-shadow: 0px -1px 12px 2px $color2;
    box-shadow: 0px -1px 12px 2px $color2;
    transform: scale(1.15);
  }
}

@media screen and (min-width: 1750px) {
  .card {
    height: 215px;
    width: 150px;
    min-width: 115px;
    p {
      font-size: 10px;
    }
    header {
      font-size: 14px;
    }
  }
}

@media screen and (max-width: 1500px) {
  .card {
    height: 170px;
    width: 115px;
    min-width: 115px;
    p {
      font-size: 8px;
    }
    header {
      font-size: 10px;
    }
  }
}

@media screen and (min-width: 1750px) {
  .card {
    height: 215px;
    width: 150px;
    min-width: 115px;
    .tag {
      font-size: 11px;
    }
    p {
      font-size: 10px;
    }
    header {
      font-size: 16px;
      padding: 0.1rem 0;
    }
    footer {
      height: 1.3rem;
    }
    footer > p {
      font-size: 15px;
      padding: 0.1rem 0 0.2rem 0;
    }
  }
}

@media screen and (max-width: 1500px) {
  .card {
    height: 175px;
    width: 120px;
    min-width: 120px;
    .tag {
      font-size: 10px;
    }
    header {
      font-size: 15px;
      padding: 0.1rem 0;
    }
    p {
      font-size: 8px;
    }
    footer {
      height: 1rem;
    }
    footer > p {
      font-size: 12px;
      margin: 0;
      padding: 0.1rem 0 0.2rem 0;
    }
  }
}
