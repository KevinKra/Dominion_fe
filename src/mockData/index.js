export const mockActionCard = {
  name: "Village",
  tags: ["+1 Card", "+2 Actions"],
  desc: "",
  category: ["Action"],
  cost: 3,
  spendingPower: 0,
  actionsProvided: 2,
  cardsToDraw: 1,
  image: "Village.jpg",
  id: 1231
};

export const mockTreasureCard = {
  name: "Copper",
  category: ["Treasure"],
  cost: 0,
  image: "Copper.jpg",
  spendingPower: 1,
  id: 1113
};

export const mockVictoryCard = {
  name: "Province",
  category: ["Victory"],
  cost: 8,
  image: "Province.jpg",
  victoryPoints: 6,
  id: 3781
};

export const mockGameState = {
  tableDeck: [
    {
      name: "Gold",
      category: ["Money"],
      cost: 6,
      victoryPoints: 0,
      spendingPower: 3,
      buyingPower: 0,
      actionsProvided: 3,
      cardsToDraw: 0,
      image: "./gold.jpg",
      desc: "",
      tags: [],
      countAvailable: 5,
      id_list: [1011, 1012, 1013, 1014, 1015]
    },
    {
      name: "Estate",
      category: ["Victory"],
      cost: 2,
      victoryPoints: 1,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "./estate.jpg",
      desc: "",
      tags: [],
      countAvailable: 8,
      id_list: [1041, 1042, 1043, 1044, 1045, 1046, 1047, 1048]
    }
  ],
  playerOrder: ["Player_1_Name", "Player_2_Name"],
  playerInfo: {
    Player_1_Name: {
      deckSize: 10,
      topCardDiscard: 0,
      handSize: 5
    },
    Player_2_Name: {
      deckSize: 10,
      topCardDiscard: 0,
      handSize: 5
    }
  },
  activePlayerName: "Player_1_Name",
  activePlayerId: 1,
  cards: [
    {
      name: "Village",
      tags: ["+1 Card", "+2 Actions"],
      desc: "",
      category: ["Action"],
      cost: 3,
      spendingPower: 0,
      actionsProvided: 2,
      cardsToDraw: 1,
      image: "Village.jpg",
      id: 1231
    },
    {
      name: "Militia",
      tags: ["+2 Gold"],
      desc: "Each other player discards down to 3 cards in his hand.",
      category: ["Action", "Attack"],
      cost: 4,
      spendingPower: 2,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Militia.jpg",
      id: 1391
    },
    {
      name: "Smithy",
      tags: ["+3 Cards"],
      desc: "",
      category: ["Action"],
      cost: 4,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 3,
      image: "Smithy.jpg",
      id: 9531
    },
    {
      name: "Market",
      tags: ["+1 Card", "+1 Action", "+1 Buy", "+1 Gold"],
      desc: "",
      category: ["Action"],
      cost: 5,
      spendingPower: 1,
      buyingPower: 1,
      actionsProvided: 1,
      cardsToDraw: 1,
      image: "Market.jpg",
      id: 1532
    },
    {
      name: "Mine",
      tags: [],
      desc:
        "Trash a treasure Card from your hand. Gain a treasure card costing up to 3 treasure more; put it in your hand.",
      category: ["Action"],
      cost: 5,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Mine.jpg",
      id: 1522
    },
    {
      name: "Remodel",
      tags: [],
      desc:
        "Trash a card from your hand. Gain a card costing up to 2 treasure more than the trashed card.",
      category: ["Action"],
      cost: 4,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Remodel.jpg",
      id: 3333
    },
    {
      name: "Cellar",
      tags: ["+1 Action"],
      desc: "Discard any number of cards. +1 Card per card discarded.",
      category: ["Action"],
      cost: 2,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 1,
      cardsToDraw: 0,
      image: "Cellar.jpg",
      id: 9831
    },
    {
      name: "Moat",
      tags: ["+2 Cards"],
      desc:
        "When another player plays an Attack card, you may reveal this from your hand. If you do, you are unaffected by that attack.",
      category: ["Action", "Reaction"],
      cost: 2,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 2,
      image: "Moat.jpg",
      id: 1938
    },
    {
      name: "WoodCutter",
      tags: ["+1 Buy", "+2 Gold"],
      desc: "",
      category: ["Action"],
      cost: 3,
      spendingPower: 2,
      buyingPower: 1,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "WoodCutter.jpg",
      id: 9781
    },
    {
      name: "WorkShop",
      tags: [],
      desc: "Gain a card costing up to 4 treasure.",
      category: ["Action"],
      cost: 3,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Workshop.jpg",
      id: 1333
    },
    {
      name: "Gold",
      category: ["Treasure"],
      cost: 6,
      image: "Gold.jpg",
      spendingPower: 3,
      id: 1515
    },
    {
      name: "Silver",
      category: ["Treasure"],
      cost: 3,
      image: "Silver.jpg",
      spendingPower: 2,
      id: 8923
    },
    {
      name: "Copper",
      category: ["Treasure"],
      cost: 0,
      image: "Copper.jpg",
      spendingPower: 1,
      id: 1113
    },
    {
      name: "Estate",
      category: ["Victory"],
      cost: 2,
      image: "Estate.jpg",
      victoryPoints: 1,
      id: 6123
    },
    {
      name: "Duchy",
      category: ["Victory"],
      cost: 5,
      image: "Duchy.jpg",
      victoryPoints: 3,
      id: 1344
    },
    {
      name: "Province",
      category: ["Victory"],
      cost: 8,
      image: "Province.jpg",
      victoryPoints: 6,
      id: 3781
    }
  ]
};

export const mockPlayerData = {
  playerId: 1,
  deck: [
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    }
  ],
  discard: [
    {
      name: "Market",
      category: "Action",
      cost: 5,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 1,
      actionsProvided: 1,
      cardsToDraw: 1,
      image: "market.jpg",
      desc: "",
      tags: ["+1 Card", "+1 Action", "+1 Buy", "+1 Gold"]
    }
  ]
};

// const boughtCards = [
//   { name: "Village", amountBought: 2 },
//   { name: "Mine", amountBought: 1 }
// ];
