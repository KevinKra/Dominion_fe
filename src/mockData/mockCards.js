export const cards = [
  {
    name: "Village",
    tags: ["+1 Card", "+2 Actions"],
    desc: "",
    type: ["Action"],
    cost: 3,
    spendingPower: 0,
    actionsProvided: 2,
    cardsToDraw: 1,
    image: "Village.jpg",
    id: 1231
  },
  {
    name: "Militia",
    tags: ["+2 Gold"],
    desc: "Each other player discards down to 3 cards in his hand.",
    type: ["Action", "Attack"],
    cost: 4,
    spendingPower: 2,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Militia.jpg",
    id: 1391
  },
  {
    name: "Smithy",
    tags: ["+3 Cards"],
    desc: "",
    type: ["Action"],
    cost: 4,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 3,
    image: "Smithy.jpg",
    id: 9531
  },
  {
    name: "Market",
    tags: ["+1 Card", "+1 Action", "+1 Buy", "+1 Gold"],
    desc: "",
    type: ["Action"],
    cost: 5,
    spendingPower: 1,
    buyingPower: 1,
    actionsProvided: 1,
    cardsToDraw: 1,
    image: "Market.jpg",
    id: 1532
  },
  {
    name: "Mine",
    tags: [],
    desc:
      "Trash a treasure Card from your hand. Gain a treasure card costing up to 3 treasure more; put it in your hand.",
    type: ["Action"],
    cost: 5,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Mine.jpg",
    id: 1522
  },
  {
    name: "Remodel",
    tags: [],
    desc:
      "Trash a card from your hand. Gain a card costing up to 2 treasure more than the trashed card.",
    type: ["Action"],
    cost: 4,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Remodel.jpg",
    id: 3333
  },
  {
    name: "Cellar",
    tags: ["+1 Action"],
    desc: "Discard any number of cards. +1 Card per card discarded.",
    type: ["Action"],
    cost: 2,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 1,
    cardsToDraw: 0,
    image: "Cellar.jpg",
    id: 9831
  },
  {
    name: "Moat",
    tags: ["+2 Cards"],
    desc:
      "When another player plays an Attack card, you may reveal this from your hand. If you do, you are unaffected by that attack.",
    type: ["Action", "Reaction"],
    cost: 2,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 2,
    image: "Moat.jpg",
    id: 1938
  },
  {
    name: "WoodCutter",
    tags: ["+1 Buy", "+2 Gold"],
    desc: "",
    type: ["Action"],
    cost: 3,
    spendingPower: 2,
    buyingPower: 1,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "WoodCutter.jpg",
    id: 9781
  },
  {
    name: "WorkShop",
    tags: [],
    desc: "Gain a card costing up to 4 treasure.",
    type: ["Action"],
    cost: 3,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Workshop.jpg",
    id: 1333
  },
  {
    name: "Gold",
    type: ["Treasure"],
    cost: 6,
    image: "Gold.jpg",
    spendingPower: 3,
    id: 1515
  },
  {
    name: "Silver",
    type: ["Treasure"],
    cost: 3,
    image: "Silver.jpg",
    spendingPower: 2,
    id: 8923
  },
  {
    name: "Copper",
    type: ["Treasure"],
    cost: 0,
    image: "Copper.jpg",
    spendingPower: 1,
    id: 1113
  },
  {
    name: "Estate",
    type: ["Victory"],
    cost: 2,
    image: "Estate.jpg",
    victoryPoints: 1,
    id: 6123
  },
  {
    name: "Duchy",
    type: ["Victory"],
    cost: 5,
    image: "Duchy.jpg",
    victoryPoints: 3,
    id: 1344
  },
  {
    name: "Province",
    type: ["Victory"],
    cost: 8,
    image: "Province.jpg",
    victoryPoints: 6,
    id: 3781
  }
];

// const boughtCards = [
//   { name: "Village", amountBought: 2 },
//   { name: "Mine", amountBought: 1 }
// ];
