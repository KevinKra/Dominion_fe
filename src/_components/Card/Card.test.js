import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { mockTreasureCard } from '../../mockData';
import { mockVictoryCard } from '../../mockData';
import { mockActionCard } from '../../mockData';

describe('<Card />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card
        name={mockActionCard.name}
        desc={mockActionCard.desc}
        category={mockActionCard.category[0]}
        image={mockActionCard.image}
        cost={mockActionCard.cost}
        id={mockActionCard.id}
        key={mockActionCard.id}
      />
    );
  });
  it('should match the mockActionCard snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should match the mockTreasureCard snapshot', () => {
    wrapper = shallow(
      <Card
        name={mockTreasureCard.name}
        desc={mockTreasureCard.desc}
        category={mockTreasureCard.category[0]}
        image={mockTreasureCard.image}
        cost={mockTreasureCard.cost}
        id={mockTreasureCard.id}
        key={mockTreasureCard.id}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should match the mockVictoryCard snapshot', () => {
    wrapper = shallow(
      <Card
        name={mockVictoryCard.name}
        desc={mockVictoryCard.desc}
        category={mockVictoryCard.category[0]}
        image={mockVictoryCard.image}
        cost={mockVictoryCard.cost}
        id={mockVictoryCard.id}
        key={mockVictoryCard.id}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should match the loading snapshot', () => {
    const loadingWrapper = shallow(<Card />);
    expect(shallowToJson(loadingWrapper)).toMatchSnapshot();
  });
});
