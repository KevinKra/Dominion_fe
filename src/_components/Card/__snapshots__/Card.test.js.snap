// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match the loading snapshot 1`] = `
<p>
  Loading
</p>
`;

exports[`<Card /> should match the mockActionCard snapshot 1`] = `
<article
  className="card"
  id={1231}
>
  <header
    style={
      Object {
        "backgroundColor": "lightGreen",
      }
    }
  >
    Village
  </header>
  <img
    alt="Village"
    src="Village.jpg"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <footer>
    <p>
      3
    </p>
    <p>
      Action
    </p>
  </footer>
</article>
`;

exports[`<Card /> should match the mockTreasureCard snapshot 1`] = `
<article
  className="card"
  id={1113}
>
  <header
    style={
      Object {
        "backgroundColor": "lightGreen",
      }
    }
  >
    Copper
  </header>
  <img
    alt="Copper"
    src="Copper.jpg"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <footer>
    <p>
      0
    </p>
    <p>
      Treasure
    </p>
  </footer>
</article>
`;

exports[`<Card /> should match the mockVictoryCard snapshot 1`] = `
<article
  className="card"
  id={3781}
>
  <header
    style={
      Object {
        "backgroundColor": "lightGreen",
      }
    }
  >
    Province
  </header>
  <img
    alt="Province"
    src="Province.jpg"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <footer>
    <p>
      8
    </p>
    <p>
      Victory
    </p>
  </footer>
</article>
`;
